src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -249,4 +281,46 @@ public static HttpResponse executeApiCall(HttpUriRequest clientRequest) {
 		return response;
 	}
 
+	public static boolean didApiReturnedWithOutErrors(String errMsg, ApiResultWrapper arw) {
+		if (arw.getStatus() != HttpStatus.SC_OK) {
+			M_log.error(errMsg);
+			return false;
+		}
+		return true;
+	}
+
+	public static String getLTICustomParam(HttpServletRequest request, String ltiParam){
+		HashMap<String, Object> customValuesMap = getLTICustomParams(request);
+		for (String LtiParamKey : customValuesMap.keySet()) {
+			if (LtiParamKey == ltiParam) {
+				return (String) customValuesMap.get(ltiParam);
+			}
+		}
+		return null;
+	}
+
+	public static String generateSisCourseId() {
+		Random rand = new Random();
+		int ranNum = rand.nextInt(40) + 1;
+		long epoch = System.currentTimeMillis();
+		String courseSisId = "ccmC" + epoch + "-" + ranNum;
+		return courseSisId;
+	}
+
+	public static String getSisSectionIDChunk(HttpServletRequest request) {
+		String courseId = getLTICustomParam(request, LTI_PARAM_CANVAS_COURSE_ID);
+		//ideally this is not the case but if at all that happens we just want some name that goes as part of sis_section_id
+		//as not hard rule that applies
+		if(courseId==null){
+			M_log.warn("Course id could not be found for making a SIS Section Id");
```

Comment:
move the warning message after line 317, and change the text to something like "Course id could not be found for making a SIS Section Id. Used current timestamp for course id: <courseId value>"